src/math_tools/partial_derivatives.hpp

```diff
@@ -0,0 +1,73 @@
+
+// SPDX-License-Identifier: MIT
+/**
+ * @file partial_derivatives.hpp
+ * File containing functions to compute the partial derivatives
+ */
+
+#pragma once
+
+#include <ddc/ddc.hpp>
+
+#include "ddc_alias_inline_functions.hpp"
+#include "ddc_aliases.hpp"
+
+
+/**
+ * @brief Compute the partial derivative of a 
+ *
+ * For a given field @f$F(X1,X2)@f$ , compute
+ * @f$\partial_{Xi} F(X1,X2)@f$ with i=1 or 2.
+ *
+ * @param[in] Xi
+ *      The given coordinate for the partial derivative.
+ *
+ * @return @f$\partial_{Xi} F(X1,X2)@f$ 
+ */
+
+template <
+        class GridXi,
+        class GridX1,
+        class GridX2,
+        class FieldX1Builder,
+        class FieldX1Evaluator>
```

Comment:
If you want to reduce the number of elements you can reduce this to:
```suggestion
template <
        class FieldX1Builder,
        class FieldX1Evaluator>
```
by extracting the DField and Xi from the Builder/Evaluator

Reply:
This would also allow us to write:
```cpp
PartialDerivative partial_deriv(builder, evaluator);
```
as the template arguments could be auto-deduced